app-drawer/app-drawer.html

```diff
@@ -218,27 +218,47 @@
           type: Boolean,
           value: false,
           reflectToAttribute: true
+        },
+
+        /**
+         * The CSS selector for the element that should receive focus when the drawer is open.
+         * By default, when the drawer opens, it focuses the first tabbable element. That is,
+         * the first element that can receive focus.
+         */
+        focusSelector: {
+          type: String,
+          value:
+            'a[href]:not([tabindex="-1"]),' +
+            'area[href]:not([tabindex="-1"]),' +
+            'input:not([disabled]):not([tabindex="-1"]),' +
+            'select:not([disabled]):not([tabindex="-1"]),' +
+            'textarea:not([disabled]):not([tabindex="-1"]),' +
+            'button:not([disabled]):not([tabindex="-1"]),' +
+            'iframe:not([tabindex="-1"]),' +
+            '[tabindex]:not([tabindex="-1"]),' +
+            '[contentEditable=true]:not([tabindex="-1"])'
         }
 
       },
 
       listeners: {
-        track: '_track',
-        transitionend: '_transitionEnd'
+        track: '_track'
       },
 
       observers: [
         'resetLayout(position)',
-        '_updateDocScroll(persistent)',
+        '_resetFocusedContent(persistent)',
         '_resetPosition(align)'
```

Comment:
Let's change this to `_resetPosition(align, isAttached)` and remove the call to `_resetPosition` inside the `attached` function.  This way we can eliminate an extra `_resetPosition` calls at startup.